eradiate/util/xarray.py

```diff
@@ -517,45 +517,53 @@ def plane(hsphere_dataarray, phi, theta_dim="theta_o", phi_dim="phi_o", drop=Fal
     Parameter ``drop`` (bool)
         If ``True``, drop azimuth angle dimension instead of making it scalar.
 
-    Returns → :class:`~xarray.DataArray`
+    Returns → :class:`~xarray.Dataset`
         Extracted plane data set for the requested azimuth angle.
     """
-    # Retrieve values for positive half-plane
-    theta_pos = hsphere_dataarray.coords[theta_dim]
-    values_pos = hsphere_dataarray.sel(**{
-        phi_dim: phi,
-        theta_dim: theta_pos,
-        "method": "nearest",
-    })
-
-    # Retrieve values for negative half-plane
-    theta_neg = hsphere_dataarray.coords[theta_dim][1:]
-    values_neg = hsphere_dataarray.sel(**{
-        phi_dim: (phi + 180.) % 360.,
-        theta_dim: theta_neg,
-        "method": "nearest",
-    })
-
-    # Transform zeniths to negative values
-    values_neg = values_neg.assign_coords({theta_dim: -theta_neg})
-    # Reorder data
-    values_neg = values_neg.loc[{theta_dim: sorted(values_neg.coords[theta_dim].values)}]
-
-    # Combine negative and positive half-planes; drop the azimuth dimension
-    # (inserted as a scalar dimension afterwards)
-    result = xr.concat((values_neg, values_pos), dim=theta_dim).drop_vars(phi_dim)
+    theta_pos = hsphere.coords[theta_dim]
+    theta_neg = hsphere.coords[theta_dim][1:]
+    ds = xr.Dataset()
+    for var_name in ['lo', 'brf', 'brdf']:
```

Comment:
You should not mention any field here: this component is supposed to handle any data, not just the data with the vars you're using here.